pkg/server/local_observer_test.go

```diff
@@ -114,3 +109,71 @@ func TestLocalObserverServer_GetFlows(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, req.Number, uint64(i))
 }
+
+type testHook struct {
+	t            *testing.T
+	ciliumDaemon interface{}
+	stopEvery    int64
+
+	seenFlows int64
+}
+
+func (h *testHook) AfterServerInit(srv serveroption.Server) error {
+	assert.Equal(h.t, srv.GetOptions().CiliumDaemon, h.ciliumDaemon)
+	return nil
+}
+
+func (h *testHook) BeforeDecoding(ctx context.Context, payload *pb.Payload) (stop bool, err error) {
+	h.seenFlows++
+	assert.Equal(h.t, payload.Time.Seconds, h.seenFlows-1)
+	if h.seenFlows%h.stopEvery == 0 {
+		stop = true
+	}
+	return stop, nil
```

Comment:
naked return here is very confusing, it reads as is this hook should stop processing

Reply:
Fair enough, I'll introduce the early return in the if body. The current is an artifact from an earlier version which had a slightly different control flow.